source/features/conversation-timeline-filter.tsx

```diff
@@ -0,0 +1,175 @@
+	const shouldHideWholeReview =
+		['showOnlyUnresolvedComments', 'showOnlyUnresolvedReviews'].includes(currentSettings) &&
+		isWholeReviewEssentiallyResolved(review);
+
+	hide(review, shouldHideWholeReview);
+
+	if (shouldHideWholeReview) {
+		return;
+	}
+
+	for (const threadContainer of select.all('.js-resolvable-timeline-thread-container[data-resolved="true"]', review)) {
+		hideWhen(
+			threadContainer,
+			'showOnlyUnresolvedReviews'
+		);
+	}
+}
+
+function processTimelineItem(item: HTMLElement): void {
+	// PR review thread
+	if (select.exists('.js-comment[id^=pullrequestreview]', item)) {
+		processReview(item);
+		return;
+	}
+
+	// Non-comment event, always hide
+	if (!select.exists('.js-comment-container', item)) {
+		hide(item, true);
+		return;
+	}
+
+	// The comment was "hidden", thefore it's considered resolved
+	if (select.exists('.rgh-preview-hidden-comments', item)) {
+		hideWhen(
+			item,
+			'showOnlyUnresolvedComments',
+			'showOnlyUnresolvedReviews'
+		);
+		return;
+	}
+
+	// Regular, unhidden comments
+	hideWhen(
+		item,
+		'showOnlyUnresolvedReviews'
+	);
+}
+
+function processPage(): void {
+	if (currentSettings === 'default') {
+		removeClassFromAll(hiddenClassName);
+	} else {
+		for (const element of select.all('.js-timeline-item')) {
+			processTimelineItem(element);
+		}
+	}
+}
+
+async function handleSelection(): Promise<void> {
+	// The event is fired before the DOM is updated. Extensions can't access the event’s `detail` where the widget would normally specify which element was selected
+	await delay(1);
+
+	currentSettings = select(`#${filterId} [aria-checked="true"]`)!.dataset.value as State;
+
+	select(`#${filterId} .reason`)!.textContent = states[currentSettings];
+
+	// `onNewComments` registers the selectors only once
+	onNewComments(processPage);
+
+	// Actually process it right now
+	processPage();
+}
+
+function createRadio(filterSettings: State): JSX.Element {
+	const label = states[filterSettings];
+	return (
+		<div
+			className="select-menu-item"
+			role="menuitemradio"
+			tabIndex={0}
+			aria-checked={filterSettings === currentSettings ? 'true' : 'false'}
+			data-value={filterSettings}
+		>
+			<CheckIcon className="select-menu-item-icon octicon octicon-check" aria-hidden="true"/>
+			<div className="select-menu-item-text">{label || 'Show all'}</div>
+		</div>
+	);
+}
+
+function addToSidebar(position: Element): void {
+	position.before(
+		sidebarItem({
+			id: filterId,
+			name: 'Conversation view options',
+			subHeader: 'Temporarily hide content',
+			handleSelection,
+			content: [
+				createRadio('default'),
+				createRadio('showOnlyComments'),
+				createRadio('showOnlyUnresolvedComments'),
+				pageDetect.isPRConversation() && createRadio('showOnlyUnresolvedReviews')
+			]
+		})
+	);
+}
+
+function init(): void {
+	observer.observe('#partial-users-participants', {
+		add: addToSidebar
+	});
+}
+
+function deinit(): void {
+	observer.disconnect();
+}
```

Comment:
Can we just inline it? (line 174)

Reply:
No need, for the same reason why we don’t inline init